include/openthread/netdiag.h

```diff
@@ -428,8 +428,6 @@ const char *otThreadGetVendorAppUrl(otInstance *aInstance);
 /**
  * Set the vendor name string.
  *
- * Requires `OPENTHREAD_CONFIG_NET_DIAG_VENDOR_INFO_SET_API_ENABLE`.
- *
```

Comment:
We should not change the OT core and/or the existing OT APIs and their config restrictions. 

The `OPENTHREAD_CONFIG_NET_DIAG_VENDOR_INFO_SET_API_ENABLE` configuration is used to indicate whether any of the `otThreadSetVendor*()` functions are provided. Platforms that don't want this feature and want to hardcode the vendor specific paramters, can disable it which then removes all API definitions and their corresponding memory storange (resulting in smaller memory footprint/firmare size). 

If you want to use these APIs under `posix` platform, then we should instead ensure `posix-config` always enables `OPENTHREAD_CONFIG_NET_DIAG_VENDOR_INFO_SET_API_ENABLE`.